agir/front/api_views.py

```diff
@@ -110,8 +110,14 @@ def get_events(self, search_term, filters, result_limit=20):
         events = events.search(search_term).distinct()
 
         # Default: get upcoming events
-        if not eventCategory and events.upcoming().count() >= result_limit:
-            events = events.upcoming()
+        if not eventCategory:
+            countUpcoming = events.upcoming().count()
+            if countUpcoming >= result_limit:
+                events = events.upcoming()
+            else:
+                # Fill with past events if results < limit
+                events_past = events.past()[: (result_limit - countUpcoming)]
+                events = events.upcoming().union(events_past)
```

Comment:
Pas sûr de comprendre. Avant on prenait que les événements à venir s'il y en avait assez, sinon on prenait les événements sans distinction de date. Pourquoi ajouter ce `else` ?

Reply:
Avec cette ancienne condition : 
`if not eventCategory and events.upcoming().count() >= result_limit:`
quand le `count < result_limit` on a éventuellement plus d'événéments passés qu'à venir.
Or puisqu'on trie par date avant de ne prendre que `result_limit` (20), les événements passés remplacent les autres.

Reply:
(C'est pour ça que je cherchais à faire une union à la base avant le refactor déjà :) )

Reply:
Ok, mais ça casse un peu l'intérêt d'avoir un `order_by` alors, non ? 
Ce ne serait pas plus logique alors de faire :
```
        if eventCategory:
            if eventCategory == self.EVENT_FILTER_PAST:
                events = events.past()
            else:
                events = events.upcoming()
        # Query
        events = events.search(search_term).distinct().order_by("-start-time")[:result_limit]
```
puis de faire juste une fonction `sorted` avec les résultats ?

Reply:
Si justement si on veut avoir la meilleure pertinence des autres tris que sont le `name` ou éventuellement d'autres à venir, on perdrait ce tri ?